cxx-sensors/src/tools/generate_cppcheck_resources.sh

```diff
@@ -1,3 +1,19 @@
 #!/bin/sh
-cppcheck --errorlist | python cppcheck_createrules.py rules > cppcheck.xml
-cppcheck --errorlist | python cppcheck_createrules.py profile > cppcheck-profile.xml
+
+# directory where cppcheck stores description (configuration) files for libraries
+# these configurations contain additional rules about undesired (obsolete, not thread-safe etc) functions
+#
+# typical location:
+# * Red Hat based linux: /usr/share/cppcheck/
+# * cppcheck sources:    cppcheck/cfg/
+CFG_DIR=/usr/share/cppcheck/
+
+CPPCHECK_LIBRARY_ARGS=""
+for CFG in ${CFG_DIR}*.cfg; do
+   [ -e "${CFG}" ] && CPPCHECK_LIBRARY_ARGS="${CPPCHECK_LIBRARY_ARGS} --library=${CFG}" || echo "no *.cfg files exist in ${CFG_DIR}; no library specific rules will be generated!"
```

Comment:
for 1.82 the final argument list looks something like that:

cppcheck --errorlist --library=avr.cfg --library=bsd.cfg --library=cppcheck-lib.cfg --library=embedded_sql.cfg --library=gnu.cfg --library=gtk.cfg --library=microsoft_sal.cfg --library=motif.cfg --library=posix.cfg --library=qt.cfg --library=sdl.cfg --library=sfml.cfg --library=std.cfg --library=windows.cfg --library=wxwidgets.cfg